packages/@tinacms/graphql/src/database/index.ts

```diff
@@ -106,29 +124,41 @@ export class Database {
   ) => {
     const { stringifiedFile, payload, keepTemplateKey } =
       await this.stringifyFile(filepath, data)
+    const tinaSchema = await this.getSchema()
+    const collection = tinaSchema.schema.collections.find(
+      (collection) => filepath.startsWith(collection.path)
+    )
+    let collectionIndexDefinitions
+    if (collection) {
+      const indexDefinitions = await this.getIndexDefinitions()
+      collectionIndexDefinitions = indexDefinitions?.[collection.name]
+    }
     if (this.store.supportsSeeding()) {
       await this.bridge.put(filepath, stringifiedFile)
     }
-    await this.store.put(filepath, payload, keepTemplateKey)
+    await this.store.put(filepath, payload, { keepTemplateKey, collection: collection.name, indexDefinitions: collectionIndexDefinitions })
   }
 
   public put = async (filepath: string, data: { [key: string]: unknown }) => {
```

Comment:
@jeffsee55 @logan-anderson just calling your attention to the changes I am making here to the put and addPendingDocument functions to lookup the collection and indexDefinitions so they can be passed to the store.put function. Not sure if this is the best way to do this.

Reply:
That makes sense for now, hopefully we can remove `addPendingDocument` soon 👍